File: Benchmarks/enterprise/python/fastapi/testcode/benchmark_test_04314.py
# SPDX-License-Identifier: Apache-2.0
from fastapi import Request
from starlette.responses import JSONResponse
import json


async def BenchmarkTest04314(request: Request):
    forwarded_ip = request.headers.get('x-forwarded-for', '')
    try:
        data = json.loads(forwarded_ip).get('value', forwarded_ip)
    except (json.JSONDecodeError, AttributeError):
        data = forwarded_ip
    return JSONResponse({'status': 'ok'}, status_code=200, headers={'X-Echo': str(data)})

File: Benchmarks/enterprise/java/spring/testcode/BenchmarkTest04314.java
// SPDX-License-Identifier: Apache-2.0
import jakarta.servlet.http.HttpServletRequest;
import jakarta.servlet.http.HttpServletResponse;
import org.springframework.web.bind.annotation.*;

@RestController
public class BenchmarkTest04314 {

    private static final org.slf4j.Logger LOG = org.slf4j.LoggerFactory.getLogger(BenchmarkTest04314.class);

    @GetMapping("/BenchmarkTest04314")
    public void BenchmarkTest04314(@RequestHeader("X-Forwarded-For") String xForwardedFor, HttpServletRequest request, HttpServletResponse response) throws Exception {
        String forwardedIp = xForwardedFor != null ? xForwardedFor : "";
        String data;
        try { data = String.valueOf(Integer.parseInt(forwardedIp)); }
        catch (NumberFormatException e) { data = forwardedIp; }
        LOG.info("request processed");
        response.setContentType("application/json");
        response.getWriter().print("{\"id\":0}");
    }
}
